content/200-concepts/100-components/02-prisma-client/051-working-with-fields/300-working-with-composite-ids-and-constraints.mdx

```diff
@@ -0,0 +1,206 @@
+metaTitle: 'Working with compound IDs and unique constraints (Concepts)'
+metaDescription: 'How to read, write, and filter by compound IDs and unique constraints.'
+tocDepth: 2
+---
+
+<TopBlock>
+
+Composite IDs and compound unique constraints can be defined in your Prisma schema using the [`@@id`](/reference/api-reference/prisma-schema-reference#id-1) <span class="api"></span> and [`@@unique`](/reference/api-reference/prisma-schema-reference#unique-1) <span class="api"></span> attributes.
+
+<Admonition type="warning">
+
+**MongoDB does not support `@@id`**<br />
+MongoDB does not support composite IDs, which means you cannot identify a model with a `@@id` attribute.
+
+</Admonition>
+
+A composite ID or compound unique constraint uses the combined values of two fields as a primary key or identifier in your database table. In the following example, the `postId` field and `userId` field are used as a composite ID for a `Like` table:
+
+```prisma highlight=22;normal
+model User {
+  id    Int    @id @default(autoincrement())
+  name  String
+  post  Post[]
+  likes Like[]
+}
+
+model Post {
+  id      Int    @id @default(autoincrement())
+  content String
+  User    User?  @relation(fields: [userId], references: [id])
+  userId  Int?
+  likes   Like[]
+}
+
+model Like {
+  postId Int
+  userId Int
+  User   User @relation(fields: [userId], references: [id])
+  Post   Post @relation(fields: [postId], references: [id])
+
+  @@id([postId, userId])
+}
+```
+
+Querying for record from the `Like` table would yield results that look like the following:
+
+```json
+{
+  "postId": 1,
+  "userId": 1
+}
+```
+
+Although there are only two fields in the response, those two fields make up a compound ID named `postId_userId`.
+
+You can also create a named compound ID or compound unique constraint by using the `@@id` or `@@unique` attributes' `name` field. For example:
+
+```prisma highlight=7;normal
+model Like {
+  postId Int
+  userId Int
+  User   User @relation(fields: [userId], references: [id])
+  Post   Post @relation(fields: [postId], references: [id])
+
+  @@id(name: "likeId", [postId, userId])
+}
+```
+
+</TopBlock>
+
+## Where you can use compound IDs and unique constraints
+
+Compound IDs and compound unique constraints can be used when working with _unique_ data.
+
+Below is a list of Prisma Client functions that accept a compound ID or compound unique constraint in the `where` filter of the query:
+
+- `findUnique`
+- `findUniqueOrThrow`
+- `delete`
+- `update`
+- `upsert`
+
+A composite ID and a composite unique constraint is also usable when creating relational data with `connect` and `connectOrCreate`.
+
+## Filtering records by a compound ID or unique constraint
+
+Although your query results will not display a compound ID or unique constraint as a field, you can use these compound values to filter your queries for unique records:
+
+```ts highlight=3-6;normal
+const like = await prisma.like.findUnique({
+  where: {
+    likeId: {
```

Comment:
I find the connection between this `likeId` and the custom name from before not strong enough. Ideally I'd like to see Prisma Client examples for _both_ scenarios: the one where a custom name is used and one where the default name is used.